### ISSUE TITLE
Possibility of using `<>` for annotations

### ISSUE THREAD

**Issue by max-sixty** [PRQL/prql#447]
Pulling this out of https://github.com/prql/prql/issues/444, since there's a lot going on there, and this might for my edification from @aljazerzen ...

We're currently thinking of using `::` for the cases where we want to annotate a type. As a reminder, type annotations in user code would be rare — they'd be used for the stdlib and some reusable functions. They could also be used to resolve some ambiguities in literals, like `datetime` or `timestamp`.

I was thinking we could use `<type>`, since we're not using that syntax already. This is used for type constraints in rust, which is most often used for generics; e.g. mostly copied from [here](https://doc.rust-lang.org/book/ch10-01-syntax.html):

```rust
struct Point<T> {
    x: T,
    y: T,
}

impl<T> Point<T> {
    fn x(&self) -> &T {
        &self.x
    }
}

impl Point<f32> {
    fn distance_from_origin(&self) -> f32 {
        (self.x.powi(2) + self.y.powi(2)).sqrt()
    }
}

fn main() {
    let p = Point { x: 5, y: 10 };
    println!("p.x = {}", p.x());
    let p = Point { x: 5.0, y: 10.0 };
    println!("distance = {}", p.distance_from_origin());
}
```

So `<>` is used for generics in `impl<T>`, but it's also used for concrete types in `Point<f32>` — this is concrete because it's not defined on the `impl`.

It's also used for concrete types in turbofish: 

```rust
let doubled = a.iter().map(|x| x * 2).collect::<Vec<i32>>();
```

So for example re:

> I would also avoid 4., because `<>` may be needed for generic types: `func sum col::column<number> -> ...`

This could be expressed as:

```
col<column<number>>
```

...and I'm not sure we give anything up, except for the possibility to name generics the same name as concrete types, which seems quite unlikely, given I imagine it's unlikely we build a very sophisticated type system? 
Can we think of examples where we would be giving something up?

Ofc:
- I may well be missing something.
- Maybe people don't like `<>` regardless — in which case totally reasonable to not use it!

**Comment by aljazerzen** [comment]
> ```rust
> let doubled = a.iter().map(|x| x * 2).collect::<Vec<i32>>();
> ```

This is actually a generic argument to function `fn collect<T>(...)`. So Rust only uses `<>` for generics.

But to be honest, if we use `:` for named args, also having `::` for type annotations is not ideal. `<>` is at least already used within types so it can be extended to type annotations.

TLDR; I agree.

**Comment by max-sixty** [comment]
OK!!

(it's v important to me that we reach these decisions in a collaborative way and weigh our views based on our contributions — but of all people I trust you to speak your mind. So sounds great, but also if you change your mind that's OK too!)

**Comment by aljazerzen** [comment]
Done, see aafb005e7786e24f778c7248e4992436d8e36dd0
